src/Cli/dotnet/commands/dotnet-workload/install/IInstaller.cs

```diff
@@ -38,14 +38,16 @@ internal interface IInstaller : IWorkloadManifestInstaller
         /// Delete the install state file at the specified path.
         /// </summary>
         /// <param name="sdkFeatureBand">The SDK feature band of the install state file.</param>
-        void DeleteInstallState(SdkFeatureBand sdkFeatureBand);
+        void DeleteInstallState(string dotnetDir, SdkFeatureBand sdkFeatureBand);
 
         /// <summary>
         /// Writes the specified JSON contents to the install state file.
         /// </summary>
         /// <param name="sdkFeatureBand">The SDK feature band of the install state file.</param>
         /// <param name="jsonLines">The JSON contents describing the install state.</param>
-        void WriteInstallState(SdkFeatureBand sdkFeatureBand, IEnumerable<string> jsonLines);
+        void WriteInstallState(string dotnetDir, SdkFeatureBand sdkFeatureBand, Dictionary<string, string> jsonLines);
+
+        void AdjustInstallMode(string dotnetDir, SdkFeatureBand sdkFeatureBand, string newMode);
```

Comment:
Nit: I think `UpdateInstallMode` would be more idiomatic than `AdjustInstallMode`.